backends/redis.go

```diff
@@ -134,58 +136,92 @@ func (r *Redis) Execute(cmd string, args ...interface{}) (interface{}, error) {
 	return client.Do(cmd, args...)
 }
 
-func (r *Redis) Delete(keys ...interface{}) int {
+func (r *Redis) Delete(keys ...interface{}) (int, error) {
 	value, err := redis.Int(r.Execute("DEL", keys...))
 	if err != nil {
-		return -1
+		return -1, err
 	}
-	return value
+	return value, nil
 }
 
-func (r *Redis) Get(key string) string {
+func (r *Redis) Get(key string) (string, error) {
 	value, err := redis.String(r.Execute("GET", key))
 	if err != nil {
-		return ""
+		return "", err
 	}
-	return value
+	return value, nil
 }
 
-func (r *Redis) Set(key string, value interface{}) bool {
+func (r *Redis) Set(key string, value interface{}) (bool, error) {
 	_, err := r.Execute("SET", key, value)
 	if err != nil {
-		return false
+		return false, err
 	}
-	return true
+	return true, nil
 }
 
-func (r *Redis) MGet(keys ...interface{}) []string {
+func (r *Redis) MGet(keys ...interface{}) ([]string, error) {
 	values, err := redis.Strings(r.Execute("MGET", keys...))
 	if err != nil {
-		return []string{}
+		return []string{}, err
 	}
-	return values
+	return values, nil
 }
 
-func (r *Redis) MSet(mapOfKeyVal map[string]interface{}) bool {
+func (r *Redis) MSet(mapOfKeyVal map[string]interface{}) (bool, error) {
 	_, err := r.Execute("MSET", redis.Args{}.AddFlat(mapOfKeyVal)...)
 	if err != nil {
-		return false
+		return false, err
 	}
-	return true
+	return true, nil
 }
 
-func (r *Redis) Expire(key string, duration int) bool {
+func (r *Redis) Expire(key string, duration int) (bool, error) {
 	_, err := r.Execute("EXPIRE", key, duration)
 	if err != nil {
-		return false
+		return false, err
 	}
-	return true
+	return true, nil
 }
 
-func (r *Redis) Setex(key string, duration int, val interface{}) bool {
+func (r *Redis) Setex(key string, duration int, val interface{}) (bool, error) {
 	_, err := r.Execute("SETEX", key, duration, val)
 	if err != nil {
-		return false
+		return false, err
+	}
+	return true, nil
+}
+
+// redis SMEMBERS implementation
+func (r *Redis) Smembers(key string) (set.Set, error) {
+	s := set.NewSet()
+	val, err := redis.Values(r.Execute("SMEMBERS", key))
+	if err != nil {
+		return s, err
+	}
+	for _, item := range(val){
+		//Redis always stores set members as string; 
+		//TODO: Check if we need to handle byte converted  with type switching.
+		it := string(item.([]byte))
+		s.Add(it)
```

Comment:
Can we just use slice instead of this set library?. Lets make it very simple for clients to use this lib.